quick_sort.py

```diff
@@ -0,0 +1,63 @@
+def quisort(uslist, lo=None, hi=None):
+    """Sort in-place an unsorted list or slice of a list
+
+    lo and hi correspond to the start and stop indices for the list slice"""
+
+    if hi is None:
+        hi = len(uslist) - 1
+    if lo is None:
+        lo = 0
+
+    def partition(uslist, lo, hi):
+        """Compare and swap values over list slice"""
+        p = uslist[hi]
+        i = lo - 1
+        j = lo
+        while j < hi:
+            if uslist[j] <= p:
+                i = i + 1
+                uslist[i], uslist[j] = uslist[j], uslist[i]
+            j += 1
+        i += 1
+        uslist[i], uslist[hi] = uslist[hi], uslist[i]
+        return i
```

Comment:
It seems a little strange to have a function defined within another function here. Any reason you did it this way instead of making a second helper function?